src/resources/views/base/inc/scripts.blade.php

```diff
@@ -62,4 +62,23 @@
     $('.nav-tabs a').on('shown.bs.tab', function (e) {
         location.hash = e.target.hash.replace("#tab_", "#");
     });
-</script>
+
+    {{-- Error frame --}}
+    @if(config('app.debug'))
+    $(document).ajaxComplete(function(e, result) {
+        if(result.responseJSON?.exception !== undefined) {
+            $.ajax({
+                url: `${document.location.origin}/admin/error-frame`,
+                method: 'POST',
+                data: result.responseJSON,
+                headers: {'Accept': 'application/text'}
```

Comment:
Lovely ❤️ Such a simple commit 😅 

Don't understand _why_ this works, but indeed it stops the endless requests loop.

However, now it won't show any debug modal at all... Tested with the same red scenario I explained earlier...

I guess the better question is... why does the error frame return 500 in this use case? Should it? I don't see why it should...

Reply:
Hi @tabacitu sorry, it was missing an explanation ...

There are 2 possible errors here;
- A soft error, where there is something wrong in the request, and those errors happen only after the request.
- And a huge one, that prevents Laravel to run, usually, auto loader loads and parses controllers, because they are added on routes for example. If a controller is missing a semicolo, the laravel app can't even boot, and will throw an ugly error. So our request to `error-frame` route will not be processed.

By adding `headers: {'Accept': 'application/text'}`, the `error-frame` request will never return another error to create the recurring calls, `error-frame` request is expecting a json, and in the case of an huge error it will now return an html, so we ignore it.

---

Now that I explained the problem, I'm thinking about two possible solutions;

1) we may be able to add a middleware, to detect these huge errors, and if the request is our `error-frame` we try to return our view — not sure if possible, but it may ...
2) on our view, we may have a "fallback", now we are rendering the error-frame, if it happens to have an ugly error, we render an iframe with it. It would not follow our error layout, but at least it would show something.

@tabacitu what do you think about it?

Reply:
Hmm... ok then... I feel like we've been focusing on the wrong things lately. Let's get back to the actual issue we're trying to fix here, instead of _how_.

**_Where_ is the user likely to have an error, when working with Backpack and AJAX? Does this implementation help him debug the error in that use case or do nothing extra?**

I think the usual suspects are:
- `App\Http\Controllers\Admin\ProductCrudController::setup()`
- `App\Http\Controllers\Admin\ProductCrudController::setupListOperation()`
- `App\Http\Controllers\Admin\ProductCrudController::setupShowOperation()`
- `App\Http\Controllers\Admin\ProductCrudController::setupCreateOperation()`
- `App\Http\Controllers\Admin\ProductCrudController::setupUpdateOperation()`
- `App\Http\Requests\ProductRequest`
- `App\Models\Product`

And it looks to me like MOST errors will be hard. And this no-iframe approach might be prettier/simpler, but... it'll cover very few (if any) of the most likely dev errors. **If we need to fall back to iFrame, I'd rather scrap the current implementation entirely and go back to showing the response in an iFrame. No custom route, no custom view, no custom css. No point in showing modal errors in two ways - that would be confusing.**

----

That being said, I don't think it'd be best to do that either.

----

Let me ask you this. **Why do we need the `error-frame` route and Controller?** Why do we make the AJAX request to that route _at all_, now that we don't re-use the Flare classes? Just to show a view? **Let's move all that rendering logic to JS instead of PHP.** Once a request has been made, errored and returned a bad HTTP code, we _always_ have something like this as the response, right?

<img width="1329" alt="Screenshot 2021-11-11 at 14 21 41" src="https://user-images.githubusercontent.com/1032474/141297221-cdf41d9e-d715-43ca-841e-077abd0bcbee.png">

Then let's just do everything in JS:
- if AJAX request failed (http status code is not 200);
- and the response looks like a parsable error (has `exception` and `message` or something);
- then use that response to construct a modal, to show the bad response; we can create a new function like `showErrorModal(exceptionResponse)` that will do just that;

No point in doing a round-trip to the server just for rendering a modal, when we already have what we want to show inside it.

If we do all of that rendering in JS, I think that'll solve all our problems:
- no more infinite loop, because we don't trigger another AJAX request at all;
- no more "wrong stack" problem because we only have the stack from that original request, no other request is made;

Don't you think that'd be better @promatik ? If so, I say let's prototype this simpler JS-constructed modal. At first, it can just dump the JSON response in a `<code></code>` block, or something that simple. If it covers all of the "usual suspects" above... then we have something - we'll consider making it pretty, by applying the design you created here.

Feels like we've somehow managed to overcomplicate this feature - let's simplify it.
Let me know if you disagree with me, or I've gotten any of it wrong.

Cheers!

Reply:
Makes total sense! Actually I think you just gave me a brilliant idea 😅
When we receive a JSON exception, we can re-ask the server for that route, but requesting the HTML view, and we simply inject that view on a modal 🤷‍♂️

I'll implement it, I think this is the holy grail ✌